napytau/gui/components/graph.py

```diff
@@ -25,60 +31,209 @@ def __init__(self, parent: "App") -> None:
 
     def update_plot(self) -> None:
         """
-        Updates the graph with the latest tau value and appearance mode.
+        Is called whenever changes on the graphs appearance should occur.
         """
         self.graph_frame = self.plot(
-            self.parent.tau.get(), customtkinter.get_appearance_mode()
+             customtkinter.get_appearance_mode()
         )
         self.graph_frame.grid(
             row=0, column=0, rowspan=2, padx=10, pady=10, sticky="nsew"
         )
         self.graph_frame.grid_propagate(False)
 
-    def plot(self, value: int, appearance: str) -> Canvas:
-        """
-        Plot the graph.
-        :param appearance: The appearance mode.
-        :param value: The value.
-        :return: The canvas.
-        """
-        # the figure that will contain the plot
-        fig = Figure(figsize=(3, 2), dpi=100, facecolor="white", edgecolor="black")
+    def plot(self, appearance: str) -> Canvas:
 
-        # setting color in dependence of appearance mode
-        if appearance == "Light":
-            main_color = "white"
-            secondary_color = "#000000"
-        else:
-            main_color = "#151515"
-            secondary_color = "#ffffff"
+        #Extracting Data:
+        y_data: list[float] = []
+        x_data: list[float] = []
+        error_data: list[float] = []
+
+        y_data_fit: list[float] = []
+        x_data_fit: list[float] = []
```

Comment:
```suggestion
        shifted_intensities: list[float] = []
        shifted_intensity_errors: list[float] = []
        distances: list[float] = []
```

Data in a name is generally a smell. Try to be more precise about what the variable will contain.